compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -540,8 +540,23 @@ data BuildLfPackageGraphArgs = BuildLfPackageGraphArgs
   }
 
 -- | The graph will have an edge from package A to package B:
---   - if A depends on B _OR_
---   - if A is a data-dependency, B is a (regular) dependency, and B doesn't depend on any data-dependencies.
+--   1. if A depends on B _OR_
+--   2. if A is a data-dependency, B is a (regular) dependency, and B doesn't
+--      depend (directly or transitively) on any data-dependencies.
+--
+--   The reason we want the second kind of edge is to ensure that the instances
```

Comment:
It's not just instances that get rewritten. Typeclass definitions get turned into re-exports of compatible typeclasses from the dependencies (`Eq` from a data-dependency becomes `Eq` in the current stdlib). Not sure how to explain that here / how much detail to get into.

Reply:
thanks! I reworded a bit, now it's "uses of type classes" that get rewritten, with a parenthetical explaining that includes (type class) definitions, instances, constraints and exports.

Reply:
Nice, thank you!